lib/controller_patches.rb

```diff
@@ -50,7 +50,8 @@ def blog_cache(cache_key, expires=4.hours)
       last_version_key = "#{cache_key}-updated"
       updated = read_fragment(last_version_key)
       @fragment_key = "#{cache_key}-#{updated}"
-      if updated.nil? || Time.parse(updated) > (Time.now + expires)
+      if updated.nil? || !fragment_exist?(@fragment_key) ||
```

Comment:
check whether the fragment actually exists - it might not (just had this happen)